x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/DateRangeBlock.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.data;
+
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.core.ReleasableIterator;
+
+/**
+ * Block that stores aggregate_metric_double values.
+ */
+public sealed interface DateRangeBlock extends Block permits DateRangeArrayBlock, ConstantNullBlock {
+    @Override
+    DateRangeBlock filter(int... positions);
+
+    @Override
+    DateRangeBlock keepMask(BooleanVector mask);
+
+    @Override
+    ReleasableIterator<? extends DateRangeBlock> lookup(IntBlock positions, ByteSizeValue targetBlockSize);
+
+    @Override
+    DateRangeBlock expand();
+
+    /**
+     * Returns {@code true} if the given blocks are equal to each other, otherwise {@code false}.
+     * Two blocks are considered equal if they have the same position count, and contain the same
+     * values (including absent null values) in the same order. This definition ensures that the
+     * equals method works properly across different implementations of the AggregateMetricDoubleBlock interface.
+     */
+    static boolean equals(DateRangeBlock lhs, DateRangeBlock rhs) {
+        if (lhs == rhs) return true;
+        if (lhs.getPositionCount() != rhs.getPositionCount()) return false;
+        return LongBlock.equals(lhs.getFromBlock(), rhs.getFromBlock()) && LongBlock.equals(lhs.getToBlock(), rhs.getToBlock());
```

Comment:
Checkstyle is going to be upset about those `if` statements. This is java! It's like a cruise ship. Very wide.